packages/drivers/routerlicious-driver/src/routerliciousResolvedUrl.ts

```diff
@@ -0,0 +1,39 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type { IResolvedUrl } from "@fluidframework/driver-definitions";
+
+/**
+ * Routerlicious extends the resolved url with additional properties to control Routerlicious-specific behaviors.
+ *
+ * @privateRemarks
+ * {@link isRouterliciousResolvedUrl} can be used to detect whether an {@link @fluidframework/driver-definitions#IResolvedUrl}
+ * is an IRouterliciousResolvedUrl.
+ *
+ * @alpha
+ */
+export interface IRouterliciousResolvedUrl extends IResolvedUrl {
+	/**
+	 * A flag to facilitate type narrowing from {@link @fluidframework/driver-definitions#IResolvedUrl} to IRouterliciousResolvedUrl.
+	 */
+	routerliciousResolvedUrl: true;
+	/**
+	 * Controls whether a newly created container will be ephemeral, which means the service will not retain it
+	 * after the collaboration session ends. Only affects
+	 * {@link @fluidframework/driver-definitions#IDocumentServiceFactory.createContainer} requests.
+	 *
+	 * @defaultValue If left undefined, treated as `false`
+	 */
+	createAsEphemeral?: boolean;
+}
+
+/**
+ * Type guard to detect if an IResolvedUrl is an IRouterliciousResolvedUrl.
+ * @alpha
```

Comment:
Since it's not exported, this function probably wants to be @internal .

And noticing that the interface is also not used anywhere by us, so I'd argue for internal too, but I suspect we do have a use case for consumers for that one? Might be good to call it out too in the PR description, if we know how it looks like.

Reply:
Good point, changing the type guard to internal in latest.

The interface will ultimately be part of the communication between the RouterliciousDocumentServiceFactory and the url resolver (similar to IOdspResolvedUrl), which is why it's alpha.  The linked PR in the description shows the usage (specifically AzureUrlResolver.ts there).

I brought this up today at API council that the attach() progression between createOdspCreateContainerRequest, IRequest, OdspUrlResolver, IResolvedUrl, IOdspResolvedUrl, etc. is not great (weak and unintuitive typing and overly chatty across API boundaries), so I might spend a little time thinking about alternatives.  If we ultimately think "customers won't really write their own URL resolvers" or even "resolved urls shouldn't be a part of the attach() call" then I think it's easier to call the interface internal or eliminate it.